### PR TITLE
fix: 🤔 pause not working in sd-carousel

### LINKED ISSUE #1250: fix: 🤔 pause not working in `sd-carousel`
## Current behavior
The current "inverted" sample shows different sd-carousel samples but all of them keep moving even though 2 of them are paused. 

## Expected behavior
Pause should work if paused is clicked and if you hover (or focusing it via keyboard) the carousel.

## Steps to reproduce
https://github.com/user-attachments/assets/06c56630-8c13-4415-8a8b-290586623ec3

## Technical Information
- Please first check the autoplay function
- As soon as you hover on the sd-carousel (in paused state) its working
- Please add documentation to Autoplay Story (Figma + Storybook)

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by Vahid1919**
## Description:

I updated the autoplay behaviour to make it consistent with user expectations. I additionally added a blur on the resume button to make the the autoplay resumes as soon the button is clicked. (It needed the user to manually click somewhere else to start). 

@coraliefeil @MartaPintoTeixeira I added some code-only documentation to the Autoplay story. This needs to be agreed on and aligned with the Figma docs. 

Closes #1250. 

## Definition of Reviewable:

- [x] Documentation is created/updated
- [x] E2E tests (features, a11y, bug fixes) are created/updated

- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/carousel/carousel.stories.ts [hunk 1]
```diff
@@ -152,12 +152,14 @@ export const Loop = {
 
 /**
  * Use the `autoplay` attribute to toggle autoplay.
+ *
+ * Autoplay is automtically paused when the user interacts with the carousel or when the pause button is clicked.
```
Comment (auroraVasconcelos) [inline packages/components/src/components/carousel/carousel.stories.ts:156]:
```suggestion
 * __Hint:__ Autoplay is automtically paused when the user interacts with the carousel or when the pause button is clicked.
```

**Comment by MartaPintoTeixeira** [comment]
Code-only description added to figma docs.

**Comment by auroraVasconcelos** [comment]
Possible bug:

When pause when the pause button is clicked everything looks fine but when paused when the user interacts with the carousel the pause button doesn't change the state (check screenshot). Is this a bug?
image

**Comment by mariohamann** [comment]
> Possible bug:
> 
> 
> 
> When pause when the pause button is clicked everything looks fine but when paused when the user interacts with the carousel the pause button doesn't change the state (check screenshot). Is this a bug?
> 
> image
> 
> 
@auroraVasconcelos Can you try to rephrase this a bit or even make a video? I don't fully understand your comment.

**Comment by smfonseca** [comment]
@mariohamann I think this is what @auroraVasconcelos is mentioning:

https://github.com/user-attachments/assets/d5685727-6a2f-49a8-ac04-55c1b6fed059

File: packages/components/src/components/carousel/carousel.stories.ts [hunk 2]
```diff
@@ -152,12 +152,14 @@ export const Loop = {
 
 /**
  * Use the `autoplay` attribute to toggle autoplay.
+ *
+ * __Hint:__ Autoplay is automtically paused when the user interacts with the carousel or when the pause button is clicked.
```
Comment (smfonseca) [inline packages/components/src/components/carousel/carousel.stories.ts:156]:
```suggestion
 * __Hint:__ Autoplay is automatically paused when the user interacts with the carousel or when the pause button is clicked.
```

**Comment by Vahid1919** [comment]
@solid-design-system/core-development please take a look :)

**Review by smfonseca — APPROVED** [review]

**Pull request merged**